packages/core/src/components/menu/menuItem.tsx

```diff
@@ -199,16 +213,11 @@ export class MenuItem extends AbstractPureComponent2<MenuItemProps & React.Ancho
                 [Classes.DISABLED]: disabled,
                 // prevent popover from closing when clicking on submenu trigger or disabled item
                 [Classes.POPOVER_DISMISS]: shouldDismissPopover && !disabled && !hasSubmenu,
-                [Classes.SELECTED]: selected || (active && intentClass === undefined),
+                [Classes.SELECTED]: active && intentClass === undefined,
```

Comment:
I'm not sure if this is correct (because I'm not sure how these classes look), needs review